### PR TITLE
eth: fix multiple checkpoint fetch logic

### PR DESCRIPTION
**Pull request by manav2401**
This PR fixes a bug that was introduced in the RFC35 forkchoice implementaiton (https://github.com/maticnetwork/bor/pull/425). When the whitelisting service tries to fetch checkpoints, it fetches them in decreasing order starting from latest checkpoint to the oldest one. So, when a new checkpoint arrives, instead of eliminating the oldest checkpoint, it eliminated the latest one. We now fetch the checkpoints in an increasing order to prevent this from happening.

### CODE REVIEW

**Comment by JekaMas** [comment]
Could you add a test for the issue?

**Comment by manav2401** [comment]
> Could you add a test for the issue?

I actually tested this on a running node. But sure, I'll add some tests for this whole function to make it more deterministic.

**Review by temaniarpit27 — APPROVED** [review]

File: eth/bor_checkpoint_verifier.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type checkpointVerifier struct {
+	verify func(ctx context.Context, handler *ethHandler, checkpoint *checkpoint.Checkpoint) (string, error)
+}
+
+func newCheckpointVerifier(verifyFn func(ctx context.Context, handler *ethHandler, checkpoint *checkpoint.Checkpoint) (string, error)) *checkpointVerifier {
+	if verifyFn == nil {
```
Comment (JekaMas) [inline eth/bor_checkpoint_verifier.go:18]: could you revert logic to reduce complexity?
like

```
if if verifyFn != nil {
    return &checkpointVerifier{verifyFn}
} 
```

File: eth/bor_checkpoint_verifier.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			}
+
+			// fetch the end checkpoint block hash
+			block, err := handler.ethAPI.GetBlockByNumber(ctx, rpc.BlockNumber(checkpoint.EndBlock.Uint64()), false)
+			if err != nil {
+				log.Debug("Failed to get end block hash of checkpoint while whitelisting", "err", err)
+				return "", errEndBlock
+			}
+
+			hash := fmt.Sprintf("%v", block["hash"])
```
Comment (JekaMas) [inline eth/bor_checkpoint_verifier.go:46]: is it equal to `hash := block["hash"].String()` ?

**Review by JekaMas — APPROVED** [review]

File: eth/bor_checkpoint_verifier.go (hunk 2 above)
Comment (manav2401) [reply eth/bor_checkpoint_verifier.go:46]: Nope. I don't think it has `String()` method.

**Review by JekaMas — APPROVED** [review]

**Comment by JekaMas** [comment]
@temaniarpit27 I believe, we need that one in v0.3.0

**Review by temaniarpit27 — APPROVED** [review]

**Pull request merged**